executor/adapter.go

```diff
@@ -106,13 +115,16 @@ func (a *statement) Exec(ctx context.Context) (ast.RecordSet, error) {
 			}
 		}
 
-		// No result fields means no Recordset.
 		defer e.Close()
 		for {
 			row, err := e.Next()
 			if err != nil {
 				return nil, errors.Trace(err)
 			}
+			// Even though we don't have result set, the row is still used to indicates if there is
+			// more work to do.
+			// For example, UPDATE statement updates a single row on one Next call, we keep calling Next until
```

Comment:
1. UPDATE statement -> the UPDATE statement
2. on one Next call -> on a Next call